File: molecular/src/main/java/com/hartwig/actin/molecular/orange/datamodel/TreatmentEvidence.java
package com.hartwig.actin.molecular.orange.datamodel;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class TreatmentEvidence {

    public abstract boolean reported();

    @Nullable
    public abstract String gene();

    @NotNull
    public abstract String event();

    @Nullable
    public abstract Integer rangeRank();

    @NotNull
    public abstract String treatment();

    public abstract boolean onLabel();

    @NotNull
    public abstract EvidenceType type();

    @NotNull
    public abstract EvidenceLevel level();

    @NotNull
    public abstract EvidenceDirection direction();

    @NotNull
    public abstract Set<String> sources();

}

File: datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/evidence/TreatmentEvidence.kt
package com.hartwig.actin.datamodel.molecular.evidence


data class TreatmentEvidence(
    val treatment: String,
    val molecularMatch: MolecularMatchDetails,
    val cancerTypeMatch: CancerTypeMatchDetails,
    val evidenceLevel: EvidenceLevel,
    val evidenceLevelDetails: EvidenceLevelDetails,
    val evidenceDirection: EvidenceDirection,
    val evidenceYear: Int,
    val efficacyDescription: String,
    val treatmentTypes: Set<String>
) {
    fun isOnLabel() = cancerTypeMatch.applicability.isOnLabel()
}
 
